codice/ddf
# DDF-4168 Adds Size enforcer plugin to fail builds for artifacts that grow too large

#### What does this PR do?
Adds size enforcer plugin defined in codice/ddf-support#54.

Many of our artifacts are already well over the default size threshold of
1Mb; for the purposes of this PR, the thresholds for those modules have been set
higher.

#### Who is reviewing it? 

@peterhuffer @kcwire @bakejeyner 

#### Select relevant component teams: 
N/A

#### Ask 2 committers to review/merge the PR and tag them here.

@lessarderic
@tbatie
@vinamartin

#### How should this be tested?


- CI will ensure that nothing is broken
- Pulling down the branch and setting the threshold for an artifact lower than its actual size should fail the build 

#### Any background context you want to provide?
#### What are the relevant tickets?
There is no Jira ticket for this. This work is being performed against a GitHub Issue.
Fixes: #3774 

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
